doc: add note for gateway AssumeRole api (#5184)
juicedata/juicefs: docs/en/guide/gateway.md

See the [MinIO official example program](https://github.com/minio/minio/blob/master/docs/sts/assume-role.go).

:::note

> Superusers who are set by environment variables cannot use AssumeRole apis; only users who are added by 'mc admin user add' can use AssumeRole apis.

Reviewer: ```suggestion
Superusers defined by environment variables cannot use AssumeRole APIs; only users added by `mc admin user add` can use AssumeRole APIs.
```
